repos/fdbt-site/src/pages/api/manageTimeRestriction.ts

```diff
@@ -0,0 +1,205 @@
+import isArray from 'lodash/isArray';
+import { NextApiResponse } from 'next';
+import { getTimeRestrictionByNameAndNoc, insertTimeRestriction } from '../../data/auroradb';
+import { removeAllWhiteSpace, removeExcessWhiteSpace, isValid24hrTimeFormat } from './apiUtils/validator';
+import { NextApiRequestWithSession, ErrorInfo, FullTimeRestriction, TimeBand } from '../../interfaces';
+import { redirectToError, redirectTo, getAndValidateNoc } from './apiUtils';
+import { updateSessionAttribute } from '../../utils/sessions';
+import { GS_TIME_RESTRICTION_ATTRIBUTE } from '../../constants/attributes';
+import { TimeRestrictionDay } from 'shared/matchingJsonTypes';
+
+export const collectInputsFromRequest = (
+    req: NextApiRequestWithSession,
+    timeRestrictionDays: TimeRestrictionDay[],
+): FullTimeRestriction[] => {
+    const fullTimeRestrictions: FullTimeRestriction[] = [];
+    timeRestrictionDays.forEach((day) => {
+        const startTimeInputs = req.body[`startTime${day}`];
+        const endTimeInputs = req.body[`endTime${day}`];
+        if (isArray(startTimeInputs) && isArray(endTimeInputs)) {
```

Comment:
Just checking would these always be both an array, or both not an array?

Reply:
Yep if start time is entered on a row then end time will be entered as `''` even if left blank. I actually think we could force these to always be arrays, even if there's only a single entry, would just mean a bit of reworking - what do you reckon?

Reply:
Nah that's fine if they are always the same 👍